Cal-CS-61A-Staff/ok
# Interface for background jobs

Resolves #820. Putting this up now for feedback.
 
To test manually:
* Start Redis
* Start the server (`./manage.py server`)
* Start a worker (`./manage.py worker`)
* Go to `/admin/course/1/jobs/test/`
* Set how long the test job will run for, and whether it will fail.
* Start the job. A new window will appear. Refresh until the output appears.

TODO:
- [x] Set up kubernetes to spin up workers
- [x] Document how to set up a job (and logging) better
- [x] Fix Travis config
- [x] Make jobs course-specific (with course-level permissions)
- [x] Page to list all jobs
- [x] Write integration tests for job interface
- [x] Sentry
- [x] Manually test a job that times out
